reflex-dom-core/src/Reflex/Dom/Builder/Immediate.hs

```diff
@@ -523,9 +537,14 @@ foreign import javascript unsafe
   extractUpTo_ :: DOM.DocumentFragment -> DOM.Node -> DOM.Node -> IO ()
 extractUpTo df s e = liftJSM $ extractUpTo_ df (toNode s) (toNode e)
 #else
-extractUpTo df s e = liftJSM $ do
-  f <- eval ("(function(df,s,e){ var x = s; var y; for(;;) { y = x['nextSibling']; df['appendChild'](x); if(e===y) { break; } x = y; } })" :: Text)
-  void $ call f f (df, s, e)
+extractUpTo df (JS.toJSVal -> s) (JS.toJSVal -> e) = liftJSM $ go s
+  where
+    go x = JS.strictEqual x e >>= \case
+      True -> pure ()
+      False -> do
+        y <- x ^. JS.js ("nextSibling" :: Text)
+        _ <- df ^. JS.js1 ("appendChild" :: Text) x
+        go (pure y)
```

Comment:
Didn't the old behavior always append at least one child and also `x` when matched? extract***UpTo***

Reply:
Oh, that's interesting. I agree that this behavior is different from what was there before; I think this is because I looked at the JS backend code instead:
https://github.com/reflex-frp/reflex-dom/blob/2baa713dfc2e247d1f93380fcab443d9255f2bcb/reflex-dom-core/src/Reflex/Dom/Builder/Immediate.hs#L521
and I think the code here *is* a faithful translation of that.

So it seems to me that there is a difference in behavior of `extractUpTo` on current `develop` (2baa713dfc2e247d1f93380fcab443d9255f2bcb). This seems to have been introduced in https://github.com/reflex-frp/reflex-dom/commit/85a3aa6a7ed601901e331ef7a9546a0d13b238c7 as a performance optimization. I guess this was/is just a bug?

Reply:
Huh? I only see whitespace changes in the JS of https://github.com/reflex-frp/reflex-dom/commit/85a3aa6a7ed601901e331ef7a9546a0d13b238c7